controllers/datamovement_controller_test.go

```diff
@@ -21,35 +21,110 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 	"os"
+	"path/filepath"
+	"time"
 
 	nnfv1alpha1 "github.com/NearNodeFlash/nnf-sos/api/v1alpha1"
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gstruct"
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// This script is used to generate output similar to `dcp`. It takes in a
+	// duration and interval that can be adjusted to meet the needs of the test.
+	// The duration is how long the fake data movement operation runs (in seconds)
+	// and the interval is the time between each progress output (using sleep).
+	//
+	// This is dumped into a temporary file and then ran as a bash script.
+	progressOutputScript = `#!/usr/bin/env bash
+
+	DURATION=$1
+	INTERVAL=$2
+	
+	echo "Faking $DURATION progress entries at an interval of $INTERVALs..."
+	
+	echo "Creating 1 files."
+	echo "Copying data."
+	
+	for (( c=1; c<$DURATION; c++ )); do
+		sleep $INTERVAL
+		percent=$(awk -v n="$c" -v p="$DURATION" 'BEGIN{printf("%i\n",n/p*100)}')
+		echo "Copied $c.000 GiB (${percent}%) in 1.001 secs (4.174 GiB/s) $(($DURATION-$c)) secs left ..."
+	done
+	
+	echo 'this is stderr' >/dev/stderr
+	
+	sleep $INTERVAL
+	echo "Copied $DURATION.000 GiB (100%) in 1.001 secs (4.174 GiB/s) done"`
 )
 
 var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
-	var dm *nnfv1alpha1.NnfDataMovement = nil
-	var srcPath, destPath string
+	var dm *nnfv1alpha1.NnfDataMovement
+	var cm *corev1.ConfigMap
+	var cmData = make(map[string]string)
+	createCm := true
+	testLabelKey := "dm-test"
```

Comment:
const (and put it inline of the `JustBeforeEach` function that uses it)

Reply:
The tests also use it for verification so it needs to be declared at a wider scope. Does putting it in the file level `const()` block make more sense?